src/MediaBox.js

```diff
@@ -1,13 +1,23 @@
-import React, { Component } from 'react';
+import { Component, cloneElement } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
+import idgen from './idgen';
 
 class MediaBox extends Component {
+  constructor(props) {
+    super(props);
+
+    this.id = props.id || `mediabox${idgen()}`;
+  }
+
   componentDidMount() {
     if (typeof M !== 'undefined') {
       const { options } = this.props;
 
-      this.instance = M.Materialbox.init(this._materialBoxed, options);
+      this.instance = M.Materialbox.init(
+        document.getElementById(this.id),
```

Comment:
I know this change was also done in a previous PR but just for my understanding, why do we favour using an `id` instead of the actual node ref?

Reply:
@CloudPower97, just for my understanding why is id better than ref node?

Reply:
Ok, I guess it's to be able to use `cloneElement`